notebooks/ir_ima_visualization/ima_visualization_and_differencing.py

```diff
@@ -0,0 +1,327 @@
+        
+    standard_dev_rhs : array of floats
+        The standard deviation of the median signal of the right side of each read. 
+        
+    diff_method: str
+        The method of finding the difference between reads. 
+        Either "instantaneous" or "cumulative".
+    
+    Returns:
+    --------
+    
+    fig: figure object
+        Panel plot with subplots showing the difference between subsequent IMA reads. 
+        Above each panel, we print the mean difference $\mu$ in the count rate over the entire image. 
+        Below each panel, we list the IMSET difference, along with the time interval between the two IMSETs.
+        The statistics in orange (on the left and right side of each panel) give the mean rate and 
+        standard deviation of each side of the image, respectively. The value in green 'delta' is the 
+        difference between the left and right side of the image. 
+        The value in white "Ratio" gives the ratio of the mean difference in orange 
+        for the left versus the right side. 
+    '''
+    
+    
+    xlabel_list = ["SCI[16-15]","SCI[15-14]","SCI[14-13]","SCI[13-12]","SCI[12-11]",
+                 "SCI[11-10]","SCI[10-9]","SCI[9-8]","SCI[8-7]","SCI[[7-6]]","SCI[6-5]",
+                "SCI[5-4]","SCI[4-3]","SCI[3-2]","SCI[2-1]"]
+    
+    fig, axarr = plt.subplots(4, 4)
+    fig.set_size_inches(40, 40)
+    fig.set_dpi(40)
+    itime = integ_time[0:-1] - integ_time[1:]
+
+    siglvl = (0.5,0.5)
+    
+    ima_j = cube[:, :, 1:]
+    inst_diff = ((ima_j*(integ_time[1:]-integ_time[0]))-\
+             (ima_j*(integ_time[0:-1]-integ_time[0])))/(integ_time[1:]-integ_time[0:-1])
+ 
+    
+    for i, ax in enumerate(axarr.reshape(-1)):
+        if (i < cube.shape[-1]-2):
+            i=i+1
+            text = ax.text(50, 500, f'{median_diff_lhs[i]:.3f}\n±\n{standard_dev_lhs[i]:.3f}', color='Orange', fontsize=30)
+            text.set_path_effects([path_effects.Stroke(linewidth=15, foreground='black'),
+                   path_effects.Normal()])
+            text = ax.text(700, 500, f'{median_diff_rhs[i]:.3f}\n±\n{standard_dev_rhs[i]:.3f}', color='Orange', fontsize=30)
+            text.set_path_effects([path_effects.Stroke(linewidth=15, foreground='black'),
+                   path_effects.Normal()])
+            text = ax.text(200, 900, f'Ratio = {median_diff_lhs[i]/median_diff_rhs[i]:.2f}', color='White', fontsize=30)
+            text.set_path_effects([path_effects.Stroke(linewidth=15, foreground='black'),
+                   path_effects.Normal()])
+            text = ax.text(300, 300, f'$\Delta = ${median_diff_lhs[i]-median_diff_rhs[i]:.2f}', color='#32CD32', fontsize=30)
+            text.set_path_effects([path_effects.Stroke(linewidth=15, foreground='black'),
+                   path_effects.Normal()])
+            
+            if diff_method == "cumulative":
+
+                diff = cube[:,:,i]-cube[:,:,i+1]
```

Comment:
Based on previous comment, `diff` is now an array of all the differences. You can delete this line and change all the `diff` to `diff[i]` in the rest of the if statement.